In [1]:
#Import the necessary libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
#import the dataset
users=pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user")

In [3]:
users.sample(10)

user_id|age|gender|occupation|zip_code
907               908|44|F|librarian|68504
539                540|28|M|engineer|91201
619                  620|18|F|writer|81648
490                  491|43|F|writer|53711
586                   587|26|M|other|14216
804                   805|27|F|other|20009
883                884|44|M|engineer|55337
874                 875|24|F|student|35802
451           452|35|M|administrator|28018
834               835|44|F|executive|11577

In [4]:
users[['user_id','age','gender','occupation','zip_code']]= users['user_id|age|gender|occupation|zip_code'].str.split('|', expand=True)
users= users.drop('user_id|age|gender|occupation|zip_code',axis=1)

In [5]:
#first 25 entries
users.head(25)

user_id age gender     occupation zip_code
0        1  24      M     technician    85711
1        2  53      F          other    94043
2        3  23      M         writer    32067
3        4  24      M     technician    43537
4        5  33      F          other    15213
5        6  42      M      executive    98101
6        7  57      M  administrator    91344
7        8  36      M  administrator    05201
8        9  29      M        student    01002
9       10  53      M         lawyer    90703
10      11  39      F          other    30329
11      12  28      F          other    06405
12      13  47      M       educator    29206
13      14  45      M      scientist    55106
14      15  49      F       educator    97301
15      16  21      M  entertainment    10309
16      17  30      M     programmer    06355
17      18  35      F          other    37212
18      19  40      M      librarian    02138
19      20  42      F      homemaker    95660
20      21  26      M         writer    30068
21      22  25      M         writer    40206
22      23  30      F         artist    48197
23      24  21      F         artist    94533
24      25  39      M       engineer    55107

In [6]:
#last 10 entries
users.tail(10)

user_id age gender     occupation zip_code
933     934  61      M       engineer    22902
934     935  42      M         doctor    66221
935     936  24      M          other    32789
936     937  48      M       educator    98072
937     938  38      F     technician    55038
938     939  26      F        student    33319
939     940  32      M  administrator    02215
940     941  20      M        student    97229
941     942  48      F      librarian    78209
942     943  22      M        student    77841

In [7]:
#dataframe is indexed by 'user_id'
users= users.set_index('user_id',drop=True)
users

age gender     occupation zip_code
user_id                                   
1        24      M     technician    85711
2        53      F          other    94043
3        23      M         writer    32067
4        24      M     technician    43537
5        33      F          other    15213
...      ..    ...            ...      ...
939      26      F        student    33319
940      32      M  administrator    02215
941      20      M        student    97229
942      48      F      librarian    78209
943      22      M        student    77841

[943 rows x 4 columns]

In [8]:
#finding the shape of the dataset
users.shape

(943, 4)

There are 943 observations & there are 5 columns in the dataset.

In [9]:
#printing the names of columns
users.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [10]:
#how is the dataset indexed
users.index

Index(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10',
       ...
       '934', '935', '936', '937', '938', '939', '940', '941', '942', '943'],
      dtype='object', name='user_id', length=943)

In [11]:
users.index.name


'user_id'

The dataset is indexed as 0,1,2,3...
and the dataframe is indexed as 1,2,3,4...ie,according to the 'user_id'

In [12]:
#datatypes of each columns
users.dtypes

age           object
gender        object
occupation    object
zip_code      object
dtype: object

All columns are of object datatype

In [13]:
#printing only the occupation column
users['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

In [14]:
#number of different occupations in the dataset
users['occupation'].unique().size

21

In [15]:
#displaying the different occupations in the dataset
users['occupation'].unique()

array(['technician', 'other', 'writer', 'executive', 'administrator',
       'student', 'lawyer', 'educator', 'scientist', 'entertainment',
       'programmer', 'librarian', 'homemaker', 'artist', 'engineer',
       'marketing', 'none', 'healthcare', 'retired', 'salesman', 'doctor'],
      dtype=object)

There are 21 different occupations out of which one is 'none' i.e no occupation.
#### Therefore, the no:of different occupations is 20.

In [16]:
# getting the frequency of each occupation
frequency = users.occupation.value_counts()
frequency

student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
lawyer            12
salesman          12
none               9
doctor             7
homemaker          7
Name: occupation, dtype: int64

The most  frequent occupation is of 'student'

In [17]:
#summary of the dataframe
users.info

<bound method DataFrame.info of         age gender     occupation zip_code
user_id                                   
1        24      M     technician    85711
2        53      F          other    94043
3        23      M         writer    32067
4        24      M     technician    43537
5        33      F          other    15213
...      ..    ...            ...      ...
939      26      F        student    33319
940      32      M  administrator    02215
941      20      M        student    97229
942      48      F      librarian    78209
943      22      M        student    77841

[943 rows x 4 columns]>

In [18]:
#summary of each column
users.describe()

age gender occupation zip_code
count   943    943        943      943
unique   61      2         21      795
top      30      M    student    55414
freq     39    670        196        9

In [19]:
#summary of the 'occupation'column
users['occupation'].describe()

count         943
unique         21
top       student
freq          196
Name: occupation, dtype: object

In [20]:
#converting the datatype of 'age' column from object to integer
users["age"] = pd.to_numeric(users["age"])

In [21]:
#mean age of users
mean1=users['age'].mean()
mean1

34.05196182396607

The mean age of users is 34.05196182396607

In [23]:
#finding the age with least occurance
users['age'].value_counts().idxmin()

7